app/assets/javascripts/services/topology_service.js

```diff
@@ -250,26 +251,33 @@ ManageIQ.angular.app.service('topologyService', ['$location', '$http', 'miqServi
       controller.icons = data.data.icons;
       if (currentSelectedKinds && (Object.keys(currentSelectedKinds).length !== Object.keys(controller.kinds).length)) {
         controller.kinds = $scope.kinds = currentSelectedKinds;
+      } else if (controller.remove_hierarchy && data.data.settings && data.data.settings.containers_max_items) {
+        var size_limit = data.data.settings.containers_max_items;
+        controller.kinds = $scope.kinds = topologyService.reduce_kinds(controller.items, controller.kinds, size_limit, controller.remove_hierarchy);
       }
     };
     var refresh = function() {
       var id;
+      var url = '';
       if ($location.absUrl().match('show/$') || $location.absUrl().match('show$')) {
-        id = '';
+        url = controller.dataUrl;
       } else {
         id = '/' + (/\/show\/(\d+)/.exec($location.absUrl())[1]);
+        url = controller.detailUrl || controller.dataUrl;
```

Comment:
Aah, no, this doesn't work.

Now you're breaking the other topology.

Whether we use container_project_topology or container_topology needs to depend on the current URL (or something explicit, not just "we're in container topology").

Reply:
I am a bit confused right now. Is there any other topology controller with detailed topology? Or is the `container_topology_controller` the only one? 

Honestly i find the whole extracting some information from URL really stupid way to determine what request should be made... 

I would much rather have it explicitly defined somewhere (especially if topology will be component in future).

Reply:
> Or is the container_topology_controller the only one?

That's the only one.


> Honestly i find the whole extracting some information from URL really stupid way to determine what request should be made...

Completely agreed, we'll need a router soon.